autopush_rs/src/db/mod.rs

```diff
@@ -223,8 +219,8 @@ impl DynamoStorage {
     ) -> MyFuture<bool> {
```

Comment:
Not able to impl trait here?

Reply:
I'll merge this then try this out, moving more of the boxing into client.rs (i think it's mostly going to be moving around vs getting rid of boxes)